ggml/src/ggml-backend.cpp

```diff
@@ -1702,6 +1709,16 @@ bool ggml_backend_sched_reserve(ggml_backend_sched_t sched, struct ggml_cgraph *
     return true;
 }
 
+static void ggml_backend_sched_optimize_graph(ggml_backend_sched_t sched, struct ggml_cgraph * graph) {
+    GGML_ASSERT(sched);
+    // Run through each backend, before splitting, giving a chance to optimize.
+    // Would be better to have each backend optimize its own split, but sched->graph
+    // gets out of sync.
+    for (int i = 0; i < sched->n_backends; i++) {
+        ggml_backend_optimize_graph(sched->backends[i], graph);
+    }
```

Comment:
This absolutely needs to be done by split, otherwise the result would be at best just whatever is the last backend called, and at worst broken if one backend makes changes incompatible with other backends.